reports/helm-releases/main.go

```diff
@@ -0,0 +1,238 @@
+		err := authenticate.SwitchContextFromS3Bucket(*bucket, *kubeconfig, *region, *ctxManager)
+		if err != nil {
+			return nil, err
+		}
+	case "live-1":
+		err := authenticate.SwitchContextFromS3Bucket(*bucket, *kubeconfig, *region, *ctxLive_1)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		fmt.Println("No cluster given")
+	}
+
+	// Get all helm releases in namespaces
+
+	helmListJson, err := executeHelmList()
+	if err != nil {
+		return nil, err
+	}
+
+	namespaces, err := getNamespaces(helmListJson)
+	if err != nil {
+		return nil, err
+	}
+
+	releases, err := getHelmReleasesInNamespaces(namespaces)
+	if err != nil {
+		return nil, err
+	}
+	return releases, nil
+
+}
+
+func executeHelmList() (string, error) {
+	cmd := exec.Command("helm", "list", "--all-namespaces", "-o", "json")
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		fmt.Println(fmt.Sprint(err) + ": " + stderr.String())
+		return "", err
+	}
+	return out.String(), nil
+}
+
+func getNamespaces(helmListJson string) ([]string, error) {
+
+	var namespaces []helmNamespace
+	json.Unmarshal([]byte(helmListJson), &namespaces)
+
+	var nsList []string
+	for ns := range namespaces {
+		nsList = append(nsList, namespaces[ns].Namespace)
+	}
+	return deduplicateList(nsList), nil
+}
+
+func getHelmReleasesInNamespaces(namespaces []string) ([]helmRelease, error) {
+	var releases []helmRelease
+	for _, ns := range namespaces {
+		release, err := helmReleasesInNamespace(ns)
+		if err != nil {
+			log.Fatalln(err.Error())
+		}
+		releases = append(releases, release...)
+	}
+	return releases, nil
+}
+
+func deduplicateList(s []string) (list []string) {
+	keys := make(map[string]bool)
+
+	for _, entry := range s {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+
+	return
+}
+
+func helmReleasesInNamespace(namespace string) ([]helmRelease, error) {
+	cmd := exec.Command("helm", "whatup", "--namespace", namespace, "-o", "json")
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	if err != nil {
+		fmt.Println(fmt.Sprint(err) + ": " + stderr.String())
+		return nil, err
+	}
+
+	var rel map[string][]helmRelease
+
+	if err := json.Unmarshal(out.Bytes(), &rel); err != nil {
+		fmt.Println("Unable to unmarshal JSON: ", err)
+		return nil, err
+	}
+
+	return rel["releases"], nil
+
+}
+
+func joinAllHelmReleases(helmReleaseLive, helmReleaseManger, helmReleaseLive_1 []helmRelease) []resourceMap {
+	var clusters []resourceMap
+
+	cluster_live := resourceMap{
+		"name": "live",
+		"apps": helmReleaseLive,
+	}
+
+	cluster_manager := resourceMap{
+		"name": "manager",
+		"apps": helmReleaseManger,
+	}
+
+	cluster_live_1 := resourceMap{
+		"name": "live-1",
+		"apps": helmReleaseLive_1,
+	}
+
+	clusters = append(clusters, cluster_live, cluster_manager, cluster_live_1)
+
+	fmt.Println(clusters)
```

Comment:
I don't see the need for a print statement here.

Reply:
Done removed fmt.Println(clusters)